web/test/spec/diaSpec.js

```diff
@@ -0,0 +1,427 @@
+describe('DIA', function( ) {
+  var page = KoskiPage()
+  var todistus = TodistusPage()
+  var opinnot = OpinnotPage()
+  var editor = opinnot.opiskeluoikeusEditor()
+  var opiskeluoikeus = OpiskeluoikeusDialog()
+  var addOppija = AddOppijaPage()
+  before(Authentication().login(), resetFixtures)
+
+  describe('Opiskeluoikeuden tiedot', function () {
+    before(page.openPage, page.oppijaHaku.searchAndSelect('151013-2195'))
+    it('näytetään oikein', function() {
+      expect(extractAsText(S('.opiskeluoikeuden-tiedot'))).to.equal(
+        'Opiskeluoikeuden voimassaoloaika : 1.9.2012 — 4.6.2016\n' +
+        'Tila 4.6.2016 Valmistunut\n' +
+        '1.9.2012 Läsnä')
+    })
+  })
+
+  describe('Valmistava DIA-vaihe', function () {
+    before(page.openPage, page.oppijaHaku.searchAndSelect('151013-2195'), opinnot.valitseSuoritus(undefined, 'Valmistava DIA-vaihe'))
+
+    describe('Oppijan suorituksissa', function () {
+      it('näytetään', function () {
+        expect(opinnot.getTutkinto()).to.equal('Valmistava DIA-vaihe')
```

Comment:
BDD-ajatus toimisi syntaktisesti paremmin, jos viimeinenkin osa (tässä "Valmistava DIA-vaihe" ja "Helsingin Saksalainen koulu") kirjoitetaan DSL:ssä esim. `it`:n sisään. Nyt esim. tämä testi näyttää siis selain-runnerilla ajettuna tältä
![screen shot 2018-12-12 at 12 22 31](https://user-images.githubusercontent.com/3884592/49863281-9eaafe80-fe08-11e8-97e9-26644759006d.png)
minkä johdosta dokumentoivasta puolesta ei saada kaikkea irti.

Reply:
Tarkensin kuvausta.